apps/core/src/hooks/stake/useValidatorInfo.ts

```diff
@@ -10,13 +15,48 @@ export function useValidatorInfo({ validatorAddress }: { validatorAddress: strin
         isError: errorValidators,
     } = useIotaClientQuery('getLatestIotaSystemState');
     const { data: rollingAverageApys } = useGetValidatorsApy();
-
+    const iotaClient = useIotaClient();
     const validatorSummary =
         system?.activeValidators.find((validator) => validator.iotaAddress === validatorAddress) ||
         null;
+    const { data: inactiveValidatorData } = useQuery({
+        queryKey: [system?.inactivePoolsId, validatorAddress],
+        async queryFn() {
+            if (!system?.inactivePoolsId || !validatorAddress) {
+                throw Error('Missing params');
+            }
+            const inactiveValidators = await iotaClient.getDynamicFields({
+                parentId: normalizeIotaAddress(system?.inactivePoolsId),
+            });
 
-    const currentEpoch = Number(system?.epoch || 0);
+            const pendingInactiveValidatorsData = await Promise.all(
+                inactiveValidators.data.map(
+                    async (validator) =>
+                        await getInactiveValidatorsData(iotaClient, validator.objectId),
+                ),
+            );
 
+            return pendingInactiveValidatorsData;
+        },
+        enabled: !!system?.inactivePoolsId && !!validatorAddress,
+        select(validators) {
+            return validators.find((validator) => validator?.validatorAddress === validatorAddress);
+        },
+    });
```

Comment:
Would be nice if we can avoid duplicating logic, can't we maybe move the inactive data fetching of validators to a separate hook?